### ISSUE TITLE
IsLooselyEqual ( x, y ) in case of date objects compared to a number

### ISSUE THREAD

**Issue by anotherCoward** [tc39/ecma262#2494]
I asked [this question on stackoverflow](https://stackoverflow.com/questions/68873627/date-object-comparision-with-primitive-value-equal-to-stored-time-value-returns ) and it doesn't seems logical, that the ToPrimivite() of a date object would return a string, if it returns in every other situation a number.

Comparing something with greater or equal and smaller or equal to another object where both conditions are true makes the test similar to LooslyEqual().

This is technically the same as DateObject == number.
```js
DateObject >= number && DateObject <= number // -> true
```
But 
```js
DateObject == number // -> false
```

The ToPrimitive() defaults for Date-Object return the ToString method, because of the spec saying "default should be treated like string would be".

The note at ToPrimitive(input [, preferredType]) (https://tc39.es/ecma262/#sec-toprimitive) says:
> Date objects treat no hint as if the hint were string.

I understand this as: It should never return a primitive string on a call from ToPrimitive().

At the Date.prototype[@@toPrimitive] (https://tc39.es/ecma262/#sec-date.prototype-@@toprimitive) part in the spec i can't find any reason, why it defaults to a string but anyway, it just says that default has to be treated like string.

> Date objects, are unique among built-in ECMAScript object in **that they treat "default" as being equivalent to "string",**

I understand this as: Handle default like string would be handled and this means, with the note from ToPrimitive(), it should return as primitive number in any case.

If a String(DateObject) is called, the ToString method is called anyway so it doesn't matter if ToPrimitive() would return a number always.

Is the note at ToPrimitive() wrong/misleading or is nearly every engine missing the note from ToPrimitive()?

**Comment by bakkot** [comment]
"treat no hint as if it were string" means "treat _the absence of a hint_ as if it were string", not "not treat any hint as if it were string".

**Comment by anotherCoward** [comment]
But it doesn't make sense compared to all other possible ways of comparing or modifing a date object with a number value. Even `dateObject + number` returns a primitive number and no string. There is no situation i can think, where a behavior like the current is required.

Why is it required to return a string if the object is holding a number and return a number in every other case?

**Comment by bakkot** [comment]
I don't know the history of that decision, I'm afraid. It's been that way since the first edition of the specification and can't change now without breaking a lot of code, though.

**Comment by jmdyck** [comment]
> It's been that way since the first edition of the specification

Yup. Back then (1997), the `[[DefaultValue]]` internal method played roughly the role that the spec now ascribes to the `@@toPrimitive` method, and the definition of `[[DefaultValue]]` in 8.6.2.6 stated:
> When  the  [[DefaultValue]]  method  of  O  is  called  with  no  hint,  then  it  behaves  as  if  the  hint  were  Number, unless  O is a Date object (see section 15.9), in which case it behaves as if the hint were String.

I don't see any explanation as to why. You might have to ask @BrendanEich.

(In [this twitter thread](https://twitter.com/awbjs/status/1142118232326725632), @allenwb describes dealing with this oddity in ES6.)
